### DISCUSSION TITLE
Update of ffmpeg version

### DISCUSSION THREAD

**Discussion by sovrgn** [DreamOfIce/ffmpeg.wasm#2] [Q&A]
Hello,

can I make a request for a version of ffmpeg > 5? There seems to be a very annoying problem of versions below that rotate images and videos to default orientation of the device that took them.

I tried reaching out to the original repo but it seems to have been abandoned by the maintainers.

**Comment by DreamOfIce** [comment]
Upgrading to FFmpeg 6 is a planned feature: https://github.com/DreamOfIce/ffmpeg.wasm#todos
But this may have to wait until some legacy issues are resolved (the parts currently marked as WIP)

**Reply by sovrgn** [reply]
Thanks for the quick reply! Is there a planned timeline for this? How do you estimate the work for the todos? I am willing to make a small donation for this endeavour, since I am working on a project that uses ffmpeg.wasm and is set to release approximately in 2 months.

**Reply by DreamOfIce** [reply]
@sovrgn  I am excited to inform you that the refactored v0.13.0 has been released, please refer to the [migration guide](https://github.com/FFmpeg-wasm/FFmpeg.wasm/blob/next/docs/migrate/v0.13.md) to upgrade. 

BTW, originally I wanted to respond to your message as soon as v0.13 released. But because I've been busy this past time, it has caused the new release to be delayed until today. I'm sorry for that. Next I will work on migrating to ffmpeg v6.

Feel free to continue the discussion if you have any questions 😄

**Comment by sovrgn** [comment]
In the meantime I figured out how to get the metadata of jpg images client side by looking for markers in the byte array. (FFMPEG couldn’t get the Orientation metadata) Then I pass the value to the ffmpeg function and made a small transpose map that rotates/flips the image to the right orientation. It gets the job done. 

I mention this just in case someone with the issue of rotated images comes across this discussion. 

I still haven’t figured out how to do it for other web friendly formats but png, webp and the others don’t have a standardized Orientation metadata + I guess the main format that should worry us among them is the jpg (since it is the most common to come directly from a camera). 

Cheers
